ntbs-service/Services/AuthorizationService.cs

```diff
@@ -115,6 +116,13 @@ public async Task<IQueryable<Notification>> FilterNotificationsByUserAsync(Claim
             return notifications;
         }
 
+        public async Task<IList<Alert>> FilterTransferAlertsFromListOfAlertsByUserAsync(ClaimsPrincipal user, IList<Alert> alerts)
+        {
+            var userTbServiceCodes = (await _userService.GetTbServicesAsync(user)).Select(s => s.Code);
+            alerts.Remove(alerts.SingleOrDefault(x => x.AlertType == AlertType.TransferRequest && !userTbServiceCodes.Contains(x.TbServiceCode)));
+            return alerts;

```

Comment:
No sure about this - this seems like it will only take out at most one alert? Is the intention not to filter out all that match this condition, perhaps like:
`alerts.RemoveAll(x => x.AlertType == AlertType.TransferRequest && !userTbServiceCodes.Contains(x.TbServiceCode));`
Also, could we just say `alerts.RemoveAll(x => !userTbServiceCodes.Contains(x.TbServiceCode)));` or is the fact that it's a particular alert type actually important?

Reply:
A tricky question - there should only ever be one transfer alert that matches this criteria but I'll change to removeall to be safe. In terms of should it apply to all alerts, transfer requests are special in that they can have a different TB service to the notification so any other alert should be accessible if they have access to the notification anyway. While it should be safe to change to the last suggestion im leaning towards the first so that down the line we don't get any unexpected behaviour!

Reply:
@conorsheehansw opting to leave it as is so we're leaving most of the filtering to just permissions on the notification dealing with permissions on most of the alerts but happy to be overruled by you here!

Reply:
As discussed due to the semi-generic nature of the method name - probably worth clearing this up.